lib/ansible/modules/cloud/azure/azure_rm_image.py

```diff
@@ -0,0 +1,291 @@
+                              resource_group=image.id.split("/")[4],
+                              managed=(not (image.storage_profile.os_disk.managed_disk is None))
+                              )
+            named_images.append(image_info)
+
+        return named_images
+
+    def capture_image(self):
+
+        try:
+            vms = self.compute_client.virtual_machines
+            vms.get(resource_group_name=self.resource_group, vm_name=self.vm_name)
+        except CloudError:
+            self.fail("VM {} not found!".format(self.vm_name))
+        except Exception as e:
+            self.fail("An exception occurred: {}".format(str(e)))
+
+        image_names = self._list_images()
+
+        found = any(elem['name'] == self.name for elem in image_names)
+
+        if not found:
+
+            images = self.compute_client.images
+            source_vm = vms.get(resource_group_name=self.resource_group, vm_name=self.vm_name).id
+            params = Image(location=self.location, source_virtual_machine=SubResource(source_vm))
+            if self.check_mode:
+                return_data = dict(name=self.name,
+                                   status="Succeeded",
+                                   location=self.location,
+                                   resource_group=self.resource_group,
+                                   changed=True)
+            else:
+                vms.deallocate(self.resource_group, self.vm_name).wait()
+                vms.generalize(self.resource_group, self.vm_name)
+                operation = images.create_or_update(resource_group_name=self.resource_group,
+                                                    image_name=self.name, parameters=params)
+                operation.wait()
+                result = operation.result()
+
+                return_data = dict(name=result.name,
+                                   status=result.provisioning_state,
+                                   location=result.location,
+                                   resource_group=self.resource_group,
+                                   changed=True)
+        else:
+            return_data = dict(name=self.name,
+                               status="Image already exists",
+                               changed=False)
+
+        return return_data
+
+    def delete_image(self):
+
+        changed = False
+        status = "Not found"
+        image_names = self._list_images()
+
+        found = any(elem['name'] == self.name for elem in image_names)
+
+        if found:
+            if self.check_mode:
+                status = "Succeeded"
+                changed = "True"
+            else:
+                images = self.compute_client.images
+                operation = images.delete(resource_group_name=self.resource_group, name=self.name)
+
+                operation.wait()
+                result = operation.result()
+                status = result.status
+
+                if status == "Succeeded":
```

Comment:
Should we fail if this doesn't return "Succeeded"?

Reply:
Yes, and I see another bug here, it deletes the image regardless of resource group. Usually the outcome is the same, but it's not what we want. Imagine this list of images:
```
- name: image1
  resource_group: group1
- name: image2
  resource_group: group2
```
And now:
```
- name: Delete image
  azure_rm_image:
    name: image1
    resource_group: group2
    state: absent
```
This will set 'found' tot True but subsequently fails to delete the image, because it really isn't there.

Same issue in `azure_rm_image_facts.py`.

Will fix.

Reply:
OK, fixed that bug.